packages/protocol/scripts/truffle/make-release.ts

```diff
@@ -269,25 +279,49 @@ module.exports = async (callback: (error?: any) => number) => {
     const released: Set<string> = new Set([])
     const proposal: ProposalTx[] = []
 
-    const release = async (contractName: string) => {
+    const release = async (contractNameIn: string) => {
+      const contractName = contractNameIn // not sure this will be needed
+
       // 0. Skip already released dependencies
       if (released.has(contractName)) {
         return
       }
-      // 1. Release all dependencies. Guarantees library addresses are canonical for linking.
-      const contractDependencies = dependencies.get(contractName)
-      for (const dependency of contractDependencies) {
-        await release(dependency)
-      }
+
       console.log('Dependencies for contract', contractName)
-      // 2. Link dependencies.
-      const contractArtifact = await artifacts.require(contractName)
-      await Promise.all(contractDependencies.map((d) => contractArtifact.link(d, addresses.get(d))))
 
-      // 3. Deploy new versions of the contract or library, if indicated by the report.
+      // const contractArtifact = await artifacts.require(contractName) // I think it won't be able to find GasPriceMinimum here
+      let contractArtifact
+
+      try {
+        contractArtifact = await artifacts.require(contractName) // I think it won't be able to find GasPriceMinimum here
+      } catch {
+        // it wasn't found in the standar artifacts folder, check if it's
+        console.log('argv.network', argv.network)
+        contractArtifact = makeTruffleContractForMigrationWithoutSingleton(
+          contractName,
+          { ...networks[argv.network], name: argv.network },
+          '0.8',
+          web3
+        )
+        // TODO WARNING: make sure there are no libraries with the same name that don't get deployed
+      }
       const shouldDeployContract = Object.keys(report.contracts).includes(contractName)
       const shouldDeployLibrary = Object.keys(report.libraries).includes(contractName)
+
       if (shouldDeployContract) {
+        // Don't try to deploy and link libraries of contracts it doesn't have to deploy
```

Comment:
Move this code in here because the tooling was deploying dependencies of contracts that it didn't need to deploy